src/BrowserFetcher.ts

```diff
@@ -219,15 +229,20 @@ export class BrowserFetcher {
     if (await existsAsync(outputPath)) return this.revisionInfo(revision);
     if (!(await existsAsync(this._downloadsFolder)))
       await mkdirAsync(this._downloadsFolder);
-    try {
-      await downloadFile(url, archivePath, progressCallback);
-      await install(archivePath, outputPath);
-    } finally {
-      if (await existsAsync(archivePath)) await unlinkAsync(archivePath);
+    if (os.arch() === 'arm64') {
+      await handleArm64();
+      return;
+    } else {
+      try {
+        await downloadFile(url, archivePath, progressCallback);
+        await install(archivePath, outputPath);
+      } finally {
+        if (await existsAsync(archivePath)) await unlinkAsync(archivePath);
+      }
+      const revisionInfo = this.revisionInfo(revision);
+      if (revisionInfo) await chmodAsync(revisionInfo.executablePath, 0o755);
+      return revisionInfo;
```

Comment:
No `else` after `return`:

```suggestion
    if (os.arch() === 'arm64') {
      await handleArm64();
      return;
    }
    try {
      await downloadFile(url, archivePath, progressCallback);
      await install(archivePath, outputPath);
    } finally {
      if (await existsAsync(archivePath)) await unlinkAsync(archivePath);
    }
    const revisionInfo = this.revisionInfo(revision);
    if (revisionInfo) await chmodAsync(revisionInfo.executablePath, 0o755);
    return revisionInfo;
```